lib/chef/cookbook_version.rb

```diff
@@ -583,7 +586,7 @@ def filenames_by_name(records)
     end
 
     def yml_filenames_by_name(records)
-      records.select { |record| record[:name] =~ /\.yml$/ }.inject({}) { |memo, record| memo[File.basename(record[:name], ".yml")] = record[:full_path]; memo }
+      records.select { |record| record[:name] =~ /\.(y[a]?ml)$/ }.inject({}) { |memo, record| memo[File.basename(record[:name], File.extname(record[:name]))] = record[:full_path]; memo }
```

Comment:
what happens if we have both default.yaml and default.yml?

Reply:
We can either fail the run; or document that one is preferable to the other, and prefer it when both are present. 

I'd rather see us fail the run -- a  de-facto default  like that can easily lead to confusion. 

But I also think we should do the same when default.yml and recipes/recipe.y[a]ml are both present; instead we just log an error and continue.

Reply:
With the latest updates, we will fail the run with a message in any case we have a same-named file with different yaml/yml extension.